app/models/user.rb

```diff
@@ -1,15 +1,44 @@
 class User < ApplicationRecord
   before_save :downcase_email
+  attr_accessor :remember_token
 
-  validates :name, :password, :password_confirmation, presence: true
-  validates :name, length: {maximum: Settings.user.user_name.max_length}
-  validates :password, length: {minimum: Settings.user.password.min_length}
+  validates :name, presence: true, length: {maximum: Settings.length.digit_255}
+  validates :password, presence: true,
+  length: {minimum: Settings.length.digit_6}, allow_nil: true
   validates :email, presence: true,
-  length: {maximum: Settings.user.email.max_length},
+  length: {maximum: Settings.length.digit_255},
   uniqueness: true, format: {with: Settings.user.email.VALID_EMAIL_REGEX}
 
   has_secure_password
 
+  class << self
+    def digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+               BCrypt::Engine::MIN_COST
+             else
+               BCrypt::Engine.cost
+             end
+      BCrypt::Password.create string, cost: cost
+    end
+
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+  end
+
+  def authenticated? remember_token
+    BCrypt::Password.new(remember_digest).is_password? remember_token
+  end
+
+  def forget
+    update_attribute :remember_digest, nil
```

Comment:
```suggestion
    update_column :remember_digest, nil
```
phân biệt `update_attribute` và `update_column` nào

Reply:
- `update_attribute` : chỉ cập nhật một thuộc tính của bản ghi, vẫn chạy callback, bỏ qua validations, cập nhật trường update_at
- `update_column` : truy cập vào DB và cập nhật trực tiếp thuộc tính, không chạy callback và bỏ qua validation, trường update_at không được cập nhật